cli/metrics/metadata/build.go

```diff
@@ -0,0 +1,192 @@
+	ping, err := apiClient.Ping(context.Background())
+	if err != nil {
+		return false, err
+	}
+	return ping.BuilderVersion == types.BuilderBuildKit, nil
+}
+
+// buildxConfigDir will look for correct configuration store path;
+// if `$BUILDX_CONFIG` is set - use it, otherwise use parent directory
+// of Docker config file (i.e. `${DOCKER_CONFIG}/buildx`)
+func buildxConfigDir(dockercfg *configfile.ConfigFile) string {
+	if buildxConfig := os.Getenv("BUILDX_CONFIG"); buildxConfig != "" {
+		return buildxConfig
+	}
+	return filepath.Join(filepath.Dir(dockercfg.Filename), "buildx")
+}
+
+// buildxDriver returns the build driver being used for the build command
+func buildxDriver(dockercfg *configfile.ConfigFile, buildArgs []string) string {
+	driver := "error"
+	configDir := buildxConfigDir(dockercfg)
+	if _, err := os.Stat(configDir); err != nil {
+		return driver
+	}
+	builder := buildxBuilder(buildArgs)
+	if len(builder) == 0 {
+		// if builder not defined in command, seek current in buildx store
+		// `${DOCKER_CONFIG}/buildx/current`
+		fileCurrent := path.Join(configDir, "current")
+		if _, err := os.Stat(fileCurrent); err != nil {
+			return driver
+		}
+		// content looks like
+		// {
+		//   "Key": "unix:///var/run/docker.sock",
+		//   "Name": "builder",
+		//   "Global": false
+		// }
+		rawCurrent, err := ioutil.ReadFile(fileCurrent)
+		if err != nil {
+			return driver
+		}
+		// unmarshal and returns `Name`
+		var obj map[string]interface{}
+		if err = json.Unmarshal(rawCurrent, &obj); err != nil {
+			return driver
+		}
+		if n, ok := obj["Name"]; ok {
+			builder = n.(string)
+			// `Name` will be empty if `default` builder is used
+			// {
+			//   "Key": "unix:///var/run/docker.sock",
+			//   "Name": "",
+			//   "Global": false
+			// }
+			if len(builder) == 0 {
+				builder = "default"
+			}
+		} else {
+			return driver
+		}
+	}
+
+	// if default builder return docker
+	if builder == "default" {
+		return "docker"
+	}
+
+	// read builder info and retrieve the current driver
+	// `${DOCKER_CONFIG}/buildx/instances/<builder>`
+	fileBuilder := path.Join(configDir, "instances", builder)
+	if _, err := os.Stat(fileBuilder); err != nil {
+		return driver
+	}
+	// content looks like
+	// {
+	//   "Name": "builder",
+	//   "Driver": "docker-container",
+	//   "Nodes": [
+	//     {
+	//       "Name": "builder0",
+	//       "Endpoint": "unix:///var/run/docker.sock",
+	//       "Platforms": null,
+	//       "Flags": null,
+	//       "ConfigFile": "",
+	//       "DriverOpts": null
+	//     }
+	//   ],
+	//   "Dynamic": false
+	// }
+	rawBuilder, err := ioutil.ReadFile(fileBuilder)
+	if err != nil {
+		return driver
+	}
+	// unmarshal and returns `Driver`
+	var obj map[string]interface{}
+	if err = json.Unmarshal(rawBuilder, &obj); err != nil {
+		return driver
+	}
+	if d, ok := obj["Driver"]; ok {
+		driver = d.(string)
+	}
+	// TODO(@crazy-max): include buildkit version being used by this driver (e.g., docker-container;0.9.2)
```

Comment:
again, be careful about adding dynamic information in source value, as this will map to different fields in looker (and different columns in the DB behind looker). if we need version information, that should probably be reported differently in the desktop heartbeat, not in CLI execution metrics.

Reply:
ah yes indeed, I have not updated this part about our conversation. will remove that comment for now.